### ISSUE TITLE
Pydantic: Model field alias lookup fails in case of a default list/dict value

### ISSUE THREAD

**Issue by odel4y** [astral-sh/ty#3874]
### Summary

Hi,

I noticed some type validation results on this Pydantic model which seem inconsistent to me. I executed `ty check minimum-example.py` on this file:

```python
from pydantic import BaseModel, Field


class Dog(BaseModel):
    name: str = Field(alias="Name")
    breed: str = Field(alias="Breed", default="Unknown")

    ears: list[str] = Field(alias="Ears")
    legs: list[str] = Field(alias="Legs", default=[])

    ancestors: dict[str, str] = Field(alias="Ancestors")
    siblings: dict[str, str] = Field(alias="Siblings", default={})


d_alias = Dog(
    Name="Fido",
    Breed="Labrador",
    Ears=["left", "right"],
    Legs=["front left", "front right", "back left", "back right"],
    Ancestors={"father": "Rex", "mother": "Bella"},
    Siblings={"brother": "Max", "sister": "Luna"},
)
```

I get the following validation errors:

```
> ty check minimum-fail-example.py

error[unknown-argument]: Argument `Legs` does not match any known parameter
  --> minimum-fail-example.py:19:5
   |
19 |     Legs=["front left", "front right", "back left", "back right"],
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |

error[unknown-argument]: Argument `Siblings` does not match any known parameter
  --> minimum-fail-example.py:21:5
   |
21 |     Siblings={"brother": "Max", "sister": "Luna"},
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |

Found 2 diagnostics
```

I use the aliases for all fields to instantiate the object. This seems to work for most of the fields. But in case there is a default list or dict value specified (fields `legs` and`siblings`), the alias lookup seems to fail. It _does_ work for the field `breed` with a default string value.

I didn't find a similar issue. [This one](https://github.com/astral-sh/ty/issues/3641) looks only distantly similar. So I created a new issue for my observation.

### Version

ty 0.0.52

**Comment by sharkdp** [comment]
Here's a standalone reproducer:

```py
from typing import Any, TypedDict, Unpack, overload
from typing_extensions import dataclass_transform

class Empty(TypedDict): ...

@overload
def field[T](default: T, *, alias: str, **extra: Unpack[Empty]) -> T: ...
@overload
def field(*, alias: str, **extra: Unpack[Empty]) -> Any: ...
def field(default: Any = None, *, alias: str, **extra: Unpack[Empty]) -> Any: ...

@dataclass_transform(field_specifiers=(field,))
class Model: ...

class C(Model):
    values: list[int] = field(alias="Values", default=[])

C(Values=[1,2])
```
https://play.ty.dev/c318cd98-168b-41b3-8b7b-daa900d333f9


It looks like the problem is actually in overload resolution. Instead of picking the first overload, due to the `**extra: Unpack[Empy]` signature, `field(alias="Values", default=[])` leads to an ambiguous overload in call resolution, and therefore we don't recognize the `alias` argument.
